ipa_room_exploration/ros/src/room_exploration_action_server.cpp

```diff
@@ -510,35 +511,53 @@ void RoomExplorationServer::exploreRoom(const ipa_building_msgs::RoomExploration
 		return;
 	}
 
+        //Quick and dirty, find better solution to fix orientation
+        std::vector<geometry_msgs::PoseStamped> exploration_path_pose_stamped(exploration_path.size());
+        std_msgs::Header header;
+        header.stamp = ros::Time::now();
+        header.frame_id = "/map";
+        for (size_t i=0; i<exploration_path.size(); ++i)
+        {
+          exploration_path_pose_stamped[i].header = header;
+          exploration_path_pose_stamped[i].header.seq = i;
+          exploration_path_pose_stamped[i].pose.position.x = exploration_path[i].x;
+          exploration_path_pose_stamped[i].pose.position.y = (room_map.rows * map_resolution) - (exploration_path[i].y - map_origin.y) + map_origin.y;
+          exploration_path_pose_stamped[i].pose.position.z = 0.;
+          Eigen::Quaterniond quaternion;
+          quaternion = Eigen::AngleAxisd((double)exploration_path[i].theta, Eigen::Vector3d::UnitZ());
+          tf::quaternionEigenToMsg(quaternion, exploration_path_pose_stamped[i].pose.orientation);
+        }
+
+        ipa_utils::reorientInDrivingDirection(&exploration_path_pose_stamped ,0);
```

Comment:
Our understanding is, that this function orients the robot towards the next pose. This is exactly what most of our algorithms deliver. If they deliver something else, this is on purpose because some algorithms care about the viewing direction of the robot especially if a camera field of view is utilized. So I would vote for using the old method as it was (i.e. completely revert the edit in this section of code). The results for footprint planning should be exactly the same. Otherwise let me know.

Reply:
After talking with ipa-fez we found that the simplest way is to flip the map vertically at the beginning of this algorithm and then all new code in this section would be obsolete. Moreover, then we could probably also get rid of the private dependency to ipa_navigation_utils.